File: 003-largest-prime-factor/largestPrimeFactor.js
function largestPrimeFactor(number) {
  let i = 2;
  let max = 1;
  while (i <= number) {
    if (number % i === 0) {
      max = i;
      number /= i;
    } else {
      i++;
    }
  }
  return max;
}

File: 003-largest-prime-factor/largestPrimeFactor.py
import sys

num = int(sys.argv[1])


def largest_prime_factor(n):
    i = 2
    max = 1
    while i <= n:
        if n % i == 0:
            max = i
            n /= i
        else:
            i += 1
    return max


print(largest_prime_factor(num))
